src/lib/model/OutboundRequestToPayModel.js

```diff
@@ -0,0 +1,490 @@
+                    // in the quote request
+                    this.data.to.fspId = payee.partyIdInfo.fspId;
+
+                    if(payee.personalInfo) {
+                        if(payee.personalInfo.complexName) {
+                            this.data.to.firstName = payee.personalInfo.complexName.firstName || this.data.to.firstName;
+                            this.data.to.middleName = payee.personalInfo.complexName.middleName || this.data.to.middleName;
+                            this.data.to.lastName = payee.personalInfo.complexName.lastName || this.data.to.lastName;
+                        }
+                        this.data.to.dateOfBirth = payee.personalInfo.dateOfBirth;
+                    }
+
+                    return resolve(payee);
+                }
+                catch(err) {
+                    return reject(err);
+                }
+            });
+
+            // set up a timeout for the resolution
+            const timeout = setTimeout(() => {
+                const err = new BackendError(`Timeout resolving payee for transfer ${this.data.transferId}`, 504);
+
+                // we dont really care if the unsubscribe fails but we should log it regardless
+                this._cache.unsubscribe(payeeKey, subId).catch(e => {
+                    this._logger.log(`Error unsubscribing (in timeout handler) ${payeeKey} ${subId}: ${e.stack || util.inspect(e)}`);
+                });
+
+                return reject(err);
+            }, this._requestProcessingTimeoutSeconds * 1000);
+
+            // now we have a timeout handler and a cache subscriber hooked up we can fire off
+            // a GET /parties request to the switch
+            try {
+                const res = await this._requests.getParties(this.data.to.idType, this.data.to.idValue,
+                    this.data.to.idSubValue);
+                this._logger.push({ peer: res }).log('Party lookup sent to peer');
+            }
+            catch(err) {
+                // cancel the timout and unsubscribe before rejecting the promise
+                clearTimeout(timeout);
+
+                // we dont really care if the unsubscribe fails but we should log it regardless
+                this._cache.unsubscribe(payeeKey, subId).catch(e => {
+                    this._logger.log(`Error unsubscribing ${payeeKey} ${subId}: ${e.stack || util.inspect(e)}`);
+                });
+
+                return reject(err);
+            }
+        });
+    }
+
+    async _executeTransactionRequest() {
+        // eslint-disable-next-line no-async-promise-executor
+        return new Promise(async (resolve, reject) => {
+            // create a transfer prepare request
+            const transactionRequest = this._buildTransactionRequest();
+
+            // listen for events on the transactionRequestId
+            const transactionRequestKey = `tr_${this.data.transactionRequestId}`;
```

Comment:
@ksatya77 see my previous comment on cache channel name. Best not to reuse "tr_..." from the outbound transfers model due to (admittedly low) possibility of ID clash.

Reply:
renamed it to txnreq_